File: week17/week17-4.py
#week17-4.py
# leedcode 896. Monotonic Array
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        #希望找到 只增加 or 減少(不可以又增加、又減少)
        N = len(nums) #有N個數字
        big, small = False, False
        for i in range(N-1): #要跑N-1次 (兩兩比較，會少一次)
            d = nums[i+1] - nums[i] 
            if d>0: big = True #變大
            if d<0: small = True #變小
        if big and small: return False
        else: return True

File: week17/week17-4.cpp
// week17-4.cpp
// leedcode 896. Monotonic Array
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        int N = nums.size(); //有N個數
        int big = 0, small=0;
        for(int i=0; i<N-1; i++) {
            int d = nums[i+1] - nums[i];
            if(d>0) big = 1;
            if(d<0) small = 1;
        }
        if(big==1 && small==1) return false;
        else return true;
    }
};